torch/nn/utils/rnn.py

```diff
@@ -123,3 +124,106 @@ def pad_packed_sequence(sequence, batch_first=False, padding_value=0.0):
     if batch_first:
         output = output.transpose(0, 1)
     return output, lengths
+
+
+def pad_sequence(sequences, lengths, batch_first=False):
+    r"""Pad a list of variable length Variables with zero
+
+    The ``pad_sequence`` pads the list of Variables on zeroth dimension and
+    stack all the sequences on zeroth dimension. For example, if the input is
+    list of sequences with size `` Lx*`` and if batch_first is False, the
+    output will be of size `` TxBx* `` and if batch_first is True,
+    output will be of size ``BxTx* ``. The ``pad_sequence`` accepts list of
+    sequences and its lengths which should be sorted in decreasing order
+
+    B is batch size
+    T is length longest sequence
+    L is length of the sequence
+    * is any trailing dimension including zero
+
+    >>> from torch.nn.utils.rnn import pad_sequence
+    >>> a = Variable(torch.ones(25, 300))
+    >>> b = Variable(torch.ones(15, 300))
+    >>> c = Variable(torch.ones(22, 300))
+    >>> pad_sequence([a, b, c], [25, 15, 22]).size()
+    torch.Size([25, 3, 300])
+
+    Note:
+        This function returns a Variable of size LxBx* or BxLx* where L is the
+        length of longest sequence (lengths[0])
+
+    Arguments:
+        sequences (list(Variable)): list of variable length sequences.
+        lengths (list[int]): list of sequences lengths of each batch element.
+        batch_first (bool, optional): if True, the input is expected in Bx*x*
+            format.
+
+    Returns:
+        Variable of size ``seq_len x len(sequences) x * ``
+            if batch_first = False
+        Variable of size ``len(sequences) x seq_len x * `` otherwise
+    """
+
+    if len(lengths) != len(sequences):
+        raise ValueError("number of elements in lengths and sequences didn't match")
+
+    out_variable = []
```

Comment:
I think it would be more efficient to pre-allocate and zero-fill that Variable, and then copy the contents of sequences into it (with appropriate slicing). This will also make it easier to make this function GPU-compatible (right now it's not, because `filler` might end up on a wrong device).

Reply:
Wouldn't the ``type_as`` in ``filler`` creation make it GPU compatible?

Reply:
It kind of will, but not completely - the tensor will be on a GPU, but on the "current GPU" set in the context, not the one of the inputs. You have to use `.new` if you want to retain the device.